src/components/TextInput.jsx

```diff
@@ -33,6 +35,7 @@ TextInput.propTypes = {
   value: PropTypes.string,
   onChange: PropTypes.func,
   type: PropTypes.oneOf(['email', 'password', 'text']),
+  key: PropTypes.string,
```

Comment:
A prop named `key` won't be forwarded to the component (`key` will always be undefined here) so you may want to rename this. Maybe something like `name` and then `key={name}`? (Naming is up to you honestly)

Reply:
I'm not sure I follow, isn't key passed to the input element?

Reply:
Sorry, I don't think I explained very well! `key` is a special prop: https://reactjs.org/warnings/special-props.html, but basically, doing `console.log({ key })` gives `{ key: undefined }` in the console.

Reply:
Ah ok! I'll fix that